Source/Core/DiscIO/VolumeDirectory.cpp

```diff
@@ -459,7 +459,7 @@ void CVolumeDirectory::WriteEntry(const File::FSTEntry& entry, u32& fstOffset, u
 		m_virtualDisk.insert(make_pair(dataOffset, entry.physicalName));
 
 		// 4 byte aligned
-		dataOffset = ROUND_UP(dataOffset + entry.size, 0x8000ull);
+		dataOffset = ROUND_UP(dataOffset + std::max(entry.size, (u64)1), 0x8000ull);
```

Comment:
Amazing how such a small change can make you look silly.  I know there was a reason i was casting in the original location i had the other code when testing things to eventually find the problem, but the fact that i did not notice that afterwards... *sigh* more sleep i guess.